contrib/babelfishpg_tsql/src/multidb.c

```diff
@@ -1328,112 +1331,181 @@ get_physical_user_name(char *db_name, char *user_name, bool suppress_error)
 	/* Truncate final result to 64 bytes */
 	truncate_tsql_identifier(result);
 
+	/* 
+	 * If the user or role is not found in the sys.babelfish_authid_user_ext 
+	 * catalog, then an error is thrown. The 'suppress_role_error' flag indicates if 
+	 * it is ok for the user or role to be absent from the catalog.
+	 */
+	if(!suppress_role_error && !user_exists_for_db(db_name, result))
+	{
+		ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("User or role \"%s\" does not exist", new_user_name)));
+	}
+
 	return result;
 }
 
-const char *
+char *
 get_dbo_schema_name(const char *dbname)
 {
+	char	   *name = palloc0(MAX_BBF_NAMEDATALEND);
+
 	if (0 == strcmp(dbname, "master"))
-		return "master_dbo";
+	{	
+		snprintf(name, MAX_BBF_NAMEDATALEND, "%s", "master_dbo");
+		return name;
```

Comment:
Instead of individually returning in every if condition, we can return at end of function. Also, we really don't need to have separate ifs for the system users/roles, we can club them into one if and do
```
snprintf(name, MAX_BBF_NAMEDATALEND, "%s_dbo", dbname);
```

This applies to all similar functions.

Reply:
Done